server/src/main/java/io/druid/client/HttpServerInventoryView.java

```diff
@@ -403,13 +387,11 @@ public boolean isStarted()
   @Override
   public boolean isSegmentLoadedByServer(String serverKey, DataSegment segment)
   {
-    synchronized (servers) {
-      DruidServerHolder holder = servers.get(serverKey);
-      if (holder != null) {
-        return holder.druidServer.getSegment(segment.getIdentifier()) != null;
-      } else {
-        return false;
-      }
+    DruidServerHolder holder = servers.get(serverKey);
+    if (holder != null) {
```

Comment:
This is a nit, but I find inverted logic like this to be hard to following when reading code.  I think it would be better as just

```
if (holder == null) {
  return false;
}

return holder.druidServer.getSegment(segment.getIdentifier()) != null;
```